src/runtime/HalideRuntime.h

```diff
@@ -490,6 +494,37 @@ typedef enum halide_type_code_t {
     #endif
 #endif
 
+/** A runtime tag for a type in the halide type system. Can be ints,
+ * unsigned ints, or floats of various bit-widths (the 'bits'
+ * field). Can also be vectors of the same (by setting the 'width'
+ * field to something larger than one). This struct is should be
+ * exactly 32-bits in size. */
+struct halide_type_t {
+    /** The basic type code: signed integer, unsigned integer, or floating point. */
+#if __cplusplus >= 201103L
+    HALIDE_ATTRIBUTE_ALIGN(1) halide_type_code_t code; // halide_type_code_t
+#else
+  HALIDE_ATTRIBUTE_ALIGN(1) unsigned int code : 8; // halide_type_code_t
+#endif
+
+    /** The number of bits of precision of a single scalar value of this type. */
+  HALIDE_ATTRIBUTE_ALIGN(1) unsigned int bits : 8;
+
+    /** How many elements (if a vector type). Should be 1 for scalar types. */
+  HALIDE_ATTRIBUTE_ALIGN(2) unsigned int width : 16;
+
+#ifdef __cplusplus
+    halide_type_t(halide_type_code_t code, uint8_t bits, uint16_t width = 1) : code(code), bits(bits), width(width) { }
```

Comment:
Why explicit bit sizes instead of uint8 and uint16?

Also, I doubt we need this c++ block. We can just use brace initialization for the first constructor, and aren't the other two functions the same as the default copy and assignment operators? Or am I missing something...